plugins/gcp/resoto_plugin_gcp/resources/compute.py

```diff
@@ -2714,13 +2720,49 @@ class GcpInstance(GcpResource, BaseInstance):
     instance_status_message: Optional[str] = field(default=None)
     instance_tags: Optional[GcpTags] = field(default=None)
 
+    def set_machine_type(self, machine_type_link_ish: str, graph_builder: GraphBuilder) -> None:
+        def find_existing_machine_type() -> Optional[GcpMachineType]:
```

Comment:
If you call this method in `post_process` than it is not safe that the related machine type is already available.
Maybe it is better to copy the relevant data in the `connect_in_graph` phase.

Reply:
Okay, agreed. We would need to put all this logic in a step that happens after all collection has been completed in order to work as intended. Putting it in `connect_in_graph()` would do that, however that would be wrong semantically, as it leaves `GcpInstance` in an incorrect / unfinished state after `post_processing()`.
If we're OK with that, I'll move it downwards.

Reply:
The only action we would do in this phase: searching the related machine type and copy cores/memory information down to every instance.
Not ideal I agree, but I think it is better this way as to introduce dependencies to make sure instance types are available before instances are fetched.